File: 백준/Silver/1026. 보물/보물.java
import java.lang.reflect.Array;
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n;

        n = sc.nextInt();

        int[] Aarray = new int[n];
        int[] Barray = new int[n];
        int[] check = new int[n];

        for (int i = 0; i < n; i++) {
            Aarray[i] = sc.nextInt();
        }

        for (int i = 0; i < n; i++) {
            Barray[i] = sc.nextInt();
        }

        Arrays.sort(Aarray);

        int sum = 0;
        for (int i = 0; i < n; i++) {
            int temp = 0;
            int idx = 0;
            for (int j = 0; j < n; j++) {
                if (temp < Barray[j] && check[j] == 0) {
                    temp = Barray[j];
                    idx = j;
                }
            }
            check[idx] = 1;
            sum = sum + (Aarray[i] * temp);
        }

        System.out.print(sum);

        sc.close();
    }
}

File: 백준/Silver/1026. 보물/보물.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;

    int Aarray[50];
    int Barray[50];
    int check[50] = { 0, };

    for (int i = 0; i < n; i++) {
        cin >> Aarray[i];
    }

    for (int i = 0; i < n; i++) {
        cin >> Barray[i];
    }

    check[n] = { 0, };

    sort(Aarray, Aarray + n);

    int sum = 0;
    for (int i = 0; i < n; i++) {
        int temp = 0;
        int idx = 0;
        for (int i = 0; i < n; i++) {
            if (temp < Barray[i] && check[i] == 0) {
                temp = Barray[i];
                idx = i;
            }
        }
        check[idx] = 1;
        sum = sum + (Aarray[i] * temp);
    }

    cout << sum;

    return 0;
}
